src/fpu/fpu_instructions.h

```diff
@@ -117,7 +117,7 @@ static double FROUND(double in){
 		// so return the closer integer within the tolerance instead
 		// of chopping to the lower value.
 		if ((fpu.cw & ExtendedPrecisionModeMask) == ExtendedPrecisionModeMask) {
-			constexpr double tolerance = 1e-14;
+			constexpr double tolerance = 1e-13;
```

Comment:
Wouldn't relative epsilon comparisons be better? You can never find a _single_ static epsilon value that would work with both very small and very large floats.

See this article:
https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/

It's an interesting and very enlightening read... depending on how much time you want to devote to the subject 😅

Reply:
Yes, great comment - this has been bugging me and I have been looking at way to do this in a performant manner, given how often this is called. We can leave this PR open while I experiment.

Reply:
Great! Glad I was able to improve this with my advice 🚀